platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -294,13 +294,56 @@ + (instancetype)mgl_predicateWithJSONObject:(id)object {
+        NSArray<NSPredicate *> *subpredicates = MGLSubpredicatesWithJSONObjects(jsonObjects);
+        if (jsonObjects.count == 2) {
+            // Determine if the expression is of BETWEEN type
+            if ([jsonObjects[0] isKindOfClass:[NSArray class]] &&
+                [jsonObjects[1] isKindOfClass:[NSArray class]]) {
+                NSArray *leftCondition = jsonObjects[0];
+                NSArray *rightCondition = jsonObjects[1];
+                NSString *leftOperator = leftCondition.firstObject;
+                NSString *rightOperator = rightCondition.firstObject;
+                
+                NSArray *limits;
+                NSExpression *leftConditionExpression;
+                
+                if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@"<="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).rightExpression, ((NSComparisonPredicate *)subpredicates[1]).rightExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).leftExpression;
+                
+                } else if ([leftOperator isEqualToString:@"<="] && [rightOperator isEqualToString:@"<="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).leftExpression, ((NSComparisonPredicate *)subpredicates[1]).rightExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).rightExpression;
+                
+                } else if([leftOperator isEqualToString:@"<="] && [rightOperator isEqualToString:@">="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).leftExpression, ((NSComparisonPredicate *)subpredicates[1]).leftExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).rightExpression;
+                
+                } else if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@">="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).rightExpression, ((NSComparisonPredicate *)subpredicates[1]).leftExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).leftExpression;
+                }
+                
+                if (limits && leftConditionExpression) {
+                     return [NSPredicate predicateWithFormat:@"%@ BETWEEN %@", leftConditionExpression, [NSExpression expressionForAggregate:limits]];
+                }
+            }
+        }
         return [NSCompoundPredicate andPredicateWithSubpredicates:subpredicates];
     }
     if ([op isEqualToString:@"any"]) {
         NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
         return [NSCompoundPredicate orPredicateWithSubpredicates:subpredicates];
     }
+    if ([op isEqualToString:@"match"]) {
```

Comment:
This case is written generically enough to be a catch-all for any operator not handled above. So we can have the code below run unconditionally and remove the assertion at the bottom of the method.